src/mds/Beacon.cc

```diff
@@ -490,9 +490,15 @@ void Beacon::notify_health(MDSRank const *mds)
   if (mds->mdcache->cache_overfull()) {
     CachedStackStringStream css;
     *css << "MDS cache is too large (" << bytes2str(mds->mdcache->cache_size())
-        << "/" << bytes2str(mds->mdcache->cache_limit_memory()) << "); "
-        << mds->mdcache->num_inodes_with_caps << " inodes in use by clients, "
-        << mds->mdcache->get_num_strays() << " stray files";
+        << "/" << bytes2str(mds->mdcache->cache_limit_memory()) << "); ";
+    // Don't include inode and stray counters in the report for standby-replay
+    // MDSs. Since it is standby-replay, both will be zero, which might
+    // confuse the users.
+    if (!mds->is_standby_replay()) {
```

Comment:
For s-r daemons, there will be a leading ";" in the warning message from the cache size string above. Suggest to clean that bit up.

Reply:
Right. Nice catch. Fixed.